### ISSUE TITLE
Warnings while cache:clear with prod environment

### ISSUE THREAD

**Issue by a-menshchikov** [symfony/symfony#41128]
**Symfony version(s) affected**: 5.3.0-BETA2

**Description**
When I'm running `bin/console cache:clear --env=prod`, I have the following output:
```
 // Clearing the cache for the prod environment with debug false                                                        

13:41:13 WARNING   [app] Failed to generate ConfigBuilder for extension FOS\RestBundle\DependencyInjection\FOSRestExtension. ["exception" => Symfony\Component\DependencyInjection\Exception\ServiceNotFoundException^ { …},"extensionClass" => "FOS\RestBundle\DependencyInjection\FOSRestExtension"]
13:41:13 WARNING   [app] Failed to generate ConfigBuilder for extension Hslavich\OneloginSamlBundle\DependencyInjection\HslavichOneloginSamlExtension. ["exception" => Symfony\Component\DependencyInjection\Exception\ServiceNotFoundException^ { …},"extensionClass" => "Hslavich\OneloginSamlBundle\DependencyInjection\HslavichOneloginSamlExtension"]
13:41:14 WARNING   [app] Failed to generate ConfigBuilder for extension Sensio\Bundle\FrameworkExtraBundle\DependencyInjection\SensioFrameworkExtraExtension. ["exception" => Symfony\Component\DependencyInjection\Exception\ServiceNotFoundException^ { …},"extensionClass" => "Sensio\Bundle\FrameworkExtraBundle\DependencyInjection\SensioFrameworkExtraExtension"]
13:41:14 WARNING   [app] Failed to generate ConfigBuilder for extension Snc\RedisBundle\DependencyInjection\SncRedisExtension. ["exception" => Symfony\Component\DependencyInjection\Exception\ServiceNotFoundException^ { …},"extensionClass" => "Snc\RedisBundle\DependencyInjection\SncRedisExtension"]
13:41:15 WARNING   [app] Failed to generate ConfigBuilder for extension Symfony\Bundle\FrameworkBundle\DependencyInjection\FrameworkExtension. ["exception" => Symfony\Component\DependencyInjection\Exception\ServiceNotFoundException^ { …},"extensionClass" => "Symfony\Bundle\FrameworkBundle\DependencyInjection\FrameworkExtension"]
13:41:15 WARNING   [app] Failed to generate ConfigBuilder for extension Symfony\Bundle\MonologBundle\DependencyInjection\MonologExtension. ["exception" => Symfony\Component\DependencyInjection\Exception\ServiceNotFoundException^ { …},"extensionClass" => "Symfony\Bundle\MonologBundle\DependencyInjection\MonologExtension"]
13:41:16 WARNING   [app] Failed to generate ConfigBuilder for extension Symfony\Bundle\SecurityBundle\DependencyInjection\SecurityExtension. ["exception" => Symfony\Component\DependencyInjection\Exception\ServiceNotFoundException^ { …},"extensionClass" => "Symfony\Bundle\SecurityBundle\DependencyInjection\SecurityExtension"]
13:41:17 WARNING   [app] Failed to generate ConfigBuilder for extension Symfony\Bundle\TwigBundle\DependencyInjection\TwigExtension. ["exception" => Symfony\Component\DependencyInjection\Exception\ServiceNotFoundException^ { …},"extensionClass" => "Symfony\Bundle\TwigBundle\DependencyInjection\TwigExtension"]
                                                                                                                        
 [OK] Cache for the "prod" environment (debug=false) was successfully cleared.                                          
```                                                                                                                        
The same time, all ConfigBuilders that used in configs from config/packages are generated (all expect TwigConfig).

**Comment by derrabus** [comment]
@Nyholm Can you have a look?

**Comment by Nyholm** [comment]
I see you get a `ServiceNotFoundException` exception. What does that one say?

**Comment by a-menshchikov** [comment]
Hmm. Debugging shows that these exceptions don't belong to the listed bundles. Inside `ConfigBuilderCacheWarmer::dumpExtension` thrown `ServiceNotFoundException` with message `You have requested a non-existent service "security.context_listener.0".`, but warning text is somewhat confusing IMO.

**Comment by Nyholm** [comment]
True, did you get any other warnings from anywhere else? 

What is the stacktrace of the `ServiceNotFoundException`?

**Comment by a-menshchikov** [comment]
There are no other warnings.
One important thing: all of this happens when there is no prod cache.

This exception thrown from https://github.com/symfony/symfony/blob/v5.3.0-BETA2/src/Symfony/Bundle/SecurityBundle/DependencyInjection/Compiler/SortFirewallListenersPass.php#L61 (`$id = 'security.context_listener.0'`). As I see, on the first run of SortFirewallListenerPass the service exists in the container definitions. It's about the following trace:
![image](https://user-images.githubusercontent.com/2580489/117447124-c35a2080-af45-11eb-87c0-9e3e7eb878de.png)

But then, when `ConfigBuilderCacheWarmer` runs, this service is no longer in the container.
![image](https://user-images.githubusercontent.com/2580489/117447360-10d68d80-af46-11eb-965a-331153054552.png)

**Comment by a-menshchikov** [comment]
It seems like this issue may be related to #41011. In both cases problem caused by non-booted kernel.

**Comment by a-menshchikov** [comment]
@Nyholm is there any news about this issue?

**Comment by Nyholm** [comment]
I think you cracked it down that it is because if #41011. 

Im not sure how to fix #41011. Maybe it is as simple as rebooting the kernel.
